api/envoy/config/listener/v3/listener.proto

```diff
@@ -45,6 +52,13 @@ message AdditionalAddress {
   // or an empty list of :ref:`socket_options <envoy_v3_api_field_config.core.v3.SocketOptionsOverride.socket_options>`,
   // it means no socket option will apply.
   core.v3.SocketOptionsOverride socket_options = 2;
+
+  // Configures TCP keepalive settings for the additional address.
+  // If specified, this will override the listener :ref:`tcp_keepalive <envoy_v3_api_field_config.listener.v3.Listener.tcp_keepalive>` configuration.
+  // If the :ref:`tcp_keepalive <envoy_v3_api_field_config.listener.v3.AdditionalAddress.TcpKeepaliveOverride.tcp_keepalive>`
+  // within the override is not specified, it means TCP keepalive settings will not be enabled
+  // for the additional address.
+  TcpKeepaliveOverride tcp_keepalive_override = 3;
```

Comment:
`SocketOptionsOverride` is necessary because we cannot determine if `repeated` field is explicitly set or not. So, we wrapped a `SocketOptionsOverride` message.

But for the TcpKeepalive, please just add a `core.v3.TcpKeepalive tcp_keepalive = 3;` here.

Reply:
I added the override for keepalive to provide a mechanism for disabling the keepalive configuration specifically at the additional address, even when it remains enabled at the listener level. For example, keepalive is enabled at the listener level, but explicitly disabled for one address:

```
name: listener
    address:
      socket_address: { address: 127.0.0.1, port_value: 1111 }
    tcp_keepalive: {} # enable keep alive for all addresses

    additional_addresses:
    - address:
        # keepalive is enable for this address by default
        socket_address: { address: 127.0.0.1, port_value: 2223 }
    - address:
        socket_address: { address: 127.0.0.1, port_value: 2222 }
        tcp_keepalive_override: {} # disable keepalive for this address
```